File: back-end/broker-service/Models/buy.cs


using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace broker.Models
{
    public class Buy
    {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int BuyId { get; set; }

        public int Quantity { get; set; }
        public string Color { get; set; }
        public int ProductId { get; set; }
        // navigational proprty
         public Broker Broker { get; set; }

        public Customer Customer { get; set; }
        




    }

}

File: app/lib/model/broker/buy.dart
class Buy {
  int? buyId;
  int? quantity;
  String? color;
  int? productId;

  Buy({this.buyId, this.quantity, this.color, this.productId});

  Buy.fromJson(Map<String, dynamic> json) {
    buyId = json['buyId'];
    quantity = json['quantity'];
    color = json['color'];
    productId = json['productId'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['buyId'] = this.buyId;
    data['quantity'] = this.quantity;
    data['color'] = this.color;
    data['productId'] = this.productId;
    return data;
  }
}